doc/tutorial/providing-autoinstall.rst

```diff
@@ -0,0 +1,83 @@
+.. _providing-autoinstall:
+
+Providing autoinstall configuration
+===================================
+
+There are two ways to provide the autoinstall configuration:
+
+* Provide :external+cloud-init:ref:`#cloud-config user data <user_data_formats-cloud_config>` containing ``autoinstall:`` configuration directives to cloud-init at boot time
+* Directly on the installation media
+
+For detailed how-to guides that provide step-by-step instructions on how to use these two methods, go to:
+
+* :ref:`Autoinstall quick start <autoinstall_quick_start>`
+* :ref:`Autoinstall quick start for s390x <autoinstall_quick_start_s390x>`
+
+
+Autoinstall by way of `cloud-config`
+------------------------------------
+
+The suggested way of providing autoinstall configuration to the Ubuntu installer is
+via cloud-init. This allows the configuration to be applied to the installer
+without having to modify the installation media.
+
+The autoinstall configuration is provided via cloud-init configuration, which is
+almost endlessly flexible. In most scenarios the easiest way will be to provide
+user data via the :external+cloud-init:ref:`datasource_nocloud` data source.
+
+When providing autoinstall via cloud-init, the autoinstall configuration is provided
+as :external+cloud-init:ref:`user_data_formats-cloud_config`. This
+means we need a :code:`#cloud-config` header. The autoinstall directives are
+placed under a top level :code:`autoinstall:` key:
+
+.. code-block:: yaml
+
+    #cloud-config
+    autoinstall:
+        version: 1
+        ....
+
+.. note::
+
+   :external+cloud-init:ref:`user_data_formats-cloud_config` files must contain
+   the ``#cloud-config`` header to be recognised as a valid cloud configuration data
+   file.
+
+
+Autoinstall on the installation media
+-------------------------------------
+
+Another option for supplying autoinstall to the Ubuntu installer is to place a
+file named :code:`autoinstall.yaml` on the installation media itself.
+
```

Comment:
I believe adding this directly here is fine because at some point in the future, it will become unnecessary to have this note (we'll just say the `autoinstall` keyword is optional for non-cloud-init).